### DISCUSSION TITLE
Is there any sample available how to use the app lifecycle events for a WinUI3.0 packaged desktop app?

### DISCUSSION THREAD

**Discussion by koenvd** [microsoft/WindowsAppSDK#1175] [Q&A] [answered]
Hey,

I'm trying to use the app lifecycle events as implemented in Reunion preview v0.8 but I'm probably getting confused by the documentation.

Basically I created a WinUI3.0 packaged desktop app referencing Reunion preview v0.8 but I don't notice any changes in behavior in comparison with the stable 8.0 version:
- If a second instance of the app is activated by the browser (based on a registered protocol handler in the manifest file) the `OnLaunched ` method in the second instance is still being called. I expected the `OnActivated ` method being called.
-  `AppInstance.FindOrRegisterInstanceForKey` is just throwing an exception in the `OnLaunched ` method but I'm not sure I'm even allowed to use that method at that point. The examples in the documentation illustrate how to use the activation event in the `winMain `function of a Win32 C++ app but this function is low level in WinUI3.0 as far as I know.   So the `OnLaunched` and `OnActivated ` methods are the way to hook into this I guess?
 
Screenshot of the exception:
![Screenshot 2021-08-04 165541](https://user-images.githubusercontent.com/15209987/128205383-6f9730ab-0aee-4354-8dbe-40de610039ad.png)

Noticed as well that lifecycle events are defined in a `Microsoft.Windows.AppLifecycle` namespace here but I can't seem to import that in the WinUI3.0 app.

So probably I'm just missing something to try this out successfully in a WinUI3.0 desktop app. Hence wondering if there's any example available that would illustrate the possibilities?

Thanks already!

**Comment by andrewleader** [comment]
@andreww-msft can you help with this AppLifecycle question?

**Reply by koenvd** [reply]
Think I was probably a bit too fast with jumping on this ship. 

Seems like the C# projections for the lifecycle API will only become available when https://github.com/microsoft/WindowsAppSDK/pull/1208 has landed.

I guess the proper way to start using this feature is now move to a C++ WinUI3.0 app desktop app? (I tried out the WindowsAppSDK1.0 preview release but now hitting https://github.com/microsoft/WindowsAppSDK/issues/1200).

Or would there be a way I could generate the projections myself and still try this out in a C# WinUI app?

A bit unrelated question but would making a C++ app have advantages on long term compared with a C# app? I guess features will only become GA ready when all projections are there as well? Or would it offer a bit more flexibiilty when new features arrive later?

**Reply by andreww-msft** [reply]
I'm in the process of writing samples for all the AppLifecycle features, and for all app types, including C++ and C#. These should be ready in the next week or so. 

As an aside, there's also work planned to ensure that the WinUI feature works correctly with other WASDK features including AppLifecycle.

**Comment by koenvd** [comment, marked as answer]
Thanks for updated @andreww-msft . Much appreciated. I'll wait for the samples then.

**Reply by koenvd** [reply]
Just FYI @andreww-msft but I was able to get this running with WindowsAppSdk 1.0 preview version with the C# projections. 

Thanks again for helping out here and of course implementing this feature!!
